src/components/loader/loader.test.js

```diff
@@ -0,0 +1,103 @@
+import * as React from "react";
+import Loader from "./loader.component";
+import Button from "../button/button.component";
+import { LOADER_SIZES } from "./loader.config";
+
+import { loader, loaderInsideButton } from "../../../cypress/locators/loader";
+
+import { positionOfElement } from "../../../cypress/support/helper";
+import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
+
+const LoaderInsideButton = ({ ...props }) => {
+  return (
+    <Button buttonType="primary" aria-label="Loading">
+      <Loader isInsideButton {...props} />
+    </Button>
+  );
+};
+
+context("Test for Loader component", () => {
+  describe("check props for Loader component", () => {
+    it.each([
+      [LOADER_SIZES[0], 12, 6],
+      [LOADER_SIZES[1], 16, 8],
+      [LOADER_SIZES[2], 20, 8],
+    ])("should render Loader using %s as size", (size, height, margin) => {
+      CypressMountWithProviders(<Loader size={size} />);
+
+      loader(positionOfElement("first"))
+        .should("have.css", "height", `${height}px`)
+        .and("have.css", "width", `${height}px`)
+        .and("have.css", "margin-right", `${margin}px`)
+        .and("have.css", "animation-delay", "0s");
+      loader(positionOfElement("second"))
+        .should("have.css", "height", `${height}px`)
+        .and("have.css", "width", `${height}px`)
+        .and("have.css", "margin-right", `${margin}px`)
+        .and("have.css", "animation-delay", "0.2s");
+      loader(positionOfElement("third"))
+        .should("have.css", "height", `${height}px`)
+        .and("have.css", "width", `${height}px`)
+        .and("have.css", "margin-right", "0px")
+        .and("have.css", "animation-delay", "0.4s");
+    });
+
+    it.each([
+      [LOADER_SIZES[0], 100],
+      [LOADER_SIZES[1], 116],
+      [LOADER_SIZES[2], 128],
+    ])("should render Loader using %s as size in Button", (size, width) => {
+      CypressMountWithProviders(<LoaderInsideButton size={size} />);
+
+      loaderInsideButton()
+        .should("have.css", "height", "40px")
+        .and("have.css", "width", `${width}px`);
+    });
+
+    it("should render Loader inside the Button component", () => {
```

Comment:
What is the purpose of this test case when there are already three preceding this one to test the Loader inside a button? The only difference I can see is that the background colour is checked. Could this assertion be added to the previous Loader in button test cases and remove this one?

Reply:
as per testing theory we suppose to check one assertion in one TS (at least the same parameters) It the test above we are checking the `size` but here only the colour :)

Reply:
Ok, could you amend the test to specify "should render Loader inside the Button component with correct background colour" then please so it's clear what its purpose is as when you see it running it looks like it's a duplicate test case from the test name.

Reply:
fixed